src/services/event-producer/sqs/sqs-event-producer-service.ts

```diff
@@ -0,0 +1,40 @@
+import 'reflect-metadata'
+import { Config } from 'node-config-ts'
+import { inject, singleton } from 'tsyringe'
+import { SQSClient, SendMessageCommand } from '@aws-sdk/client-sqs'
+import { EventProducerService } from '../event-producer-service.js'
+import { v4 as uuidv4 } from 'uuid'
+
+@singleton()
+export class SQSEventProducerService implements EventProducerService {
+  private readonly sqsClient: SQSClient
+  constructor(@inject('config') private config?: Config) {
+    this.sqsClient = new SQSClient({
+      region: process.env.AWS_REGION,
+      credentials: {
+        accessKeyId: process.env.AWS_ACCESS_KEY_ID,
+        secretAccessKey: process.env.AWS_SECRET_ACCESS_KEY,
+      },
+    })
+  }
+
+  /**
+   * Emits an anchor event by sending a message to the configured SQS
+   */
+  public async emitAnchorEvent(): Promise<void> {
+    await this.sqsClient.send(
+      new SendMessageCommand({
+        MessageBody: uuidv4().toString(),
+        MessageGroupId: 'anchor',
+        QueueUrl: process.env.AWS_SQS_URL,
```

Comment:
Config is not respected. Is it ok?

Reply:
Could you elaborate pls =)

Reply:
For sure. Sorry for that. You are passing the config into the class, but this SQS queue identifier is still from the env variable. This makes me a little bit nervous, as it splits configuration between env variables and config file, thus making deployment more complicated.

The env variables above are handled by the AWS library AFAIK. This one - I am not sure. _Maybe_ it makes sense to have it in the configuration file.

Reply:
rerequested review to look at config changes